File: hw/vaca/u3/MazeGeneratorJava/src/main/java/ec/edu/espe/mazegenerator/controller/MazeRendererASCII.java
package ec.edu.espe.mazegenerator.controller;

import ec.edu.espe.mazegenerator.model.Maze;
import ec.edu.espe.mazegenerator.model.MazeRoom;

/**
 *
 * @author Kevin Vaca Edison's OOP ESPE
 */
public class MazeRendererASCII implements MazeRenderer{

    @Override
    public void render(Maze maze) {
        int w = maze.getWidth();
        int h = maze.getHeight();

        for (int y = 0; y < h; y++) {
            for (int x = 0; x < w; x++) {
                System.out.print("+");
                System.out.print(maze.getRoom(x, y).isOpen(MazeRoom.Direction.NORTH) ? "   " : "---");
            }
            System.out.println("+");

            for (int x = 0; x < w; x++) {
                System.out.print(maze.getRoom(x, y).isOpen(MazeRoom.Direction.WEST) ? " " : "|");
                if (x == 0 && y == 0) System.out.print(" S "); // Start
                else if (x == w - 1 && y == h - 1) System.out.print(" E "); // Exit
                else System.out.print("   ");
            }
            System.out.println("|");
        }

        for (int x = 0; x < w; x++) {
            System.out.print("+---");
        }
        System.out.println("+");
    }
}

File: hw/vaca/u3/MazeGeneratorJs/MazeRendererASCII.js
import { MazeRoom } from "./MazeRoom.js";

export class MazeRendererASCII {
  render(maze) {
    const w = maze.getWidth();
    const h = maze.getHeight();

    for (let y = 0; y < h; y++) {
      let top = "", middle = "";

      for (let x = 0; x < w; x++) {
        top += "+";
        top += maze.getRoom(x, y).isOpen(MazeRoom.Direction.NORTH) ? "   " : "---";

        middle += maze.getRoom(x, y).isOpen(MazeRoom.Direction.WEST) ? " " : "|";
        if (x === 0 && y === 0) middle += " S ";
        else if (x === w - 1 && y === h - 1) middle += " E ";
        else middle += "   ";
      }

      console.log(top + "+");
      console.log(middle + "|");
    }

    console.log("+---".repeat(w) + "+");
  }
}
